### ISSUE TITLE
UpdatePolicy does not work with AutoScalingGroup Mixed Instance Policy

### ISSUE THREAD

**Issue by RyanFrench** [aws-cloudformation/aws-cloudformation-coverage-roadmap#406]
## 2. Scope of request 

AWS:: AutoScaling:: Group::MixedInstancePolicy causes update failures with UpdatePolicy when a server is launched with Weighted Capacity > 1, and using the Resource Signals feature.

## 3. Expected behavior

When updating a stack with a Mixed Instance Policy, if any instance is launched with a weighted capacity > 1, CloudFormation should take this into account when waiting for resource signals.

Feel free to describe other desired and relevant implementation details. 

## 4. Suggest specific test cases

1.
- Deploy a stack with 12 m5.large instances using CloudFormation, and a weighted capacity of 1, with an UpdatePolicy that specifies waiting for resource signals.
- Deploy an update to this stack, switching the 12x m5.large to 6x m5.xlarge, with a weighted capacity of 2.
- The stack will fail to update as it expects 12 resource signals to be sent, even though only 6 new servers have been started by the ASG.

## 6. Category (required) - Will help with tagging and be easier to find by other users to +1

1. Compute (EC2, ECS, EKS, Lambda...)

**Comment by mrapczynski** [comment]
@luiseduardocolon We have experienced something very similar to this when using the mixed instances policy on an autoscaling group. I opened a support case on in 2019, and the AWS representatives confirmed the bug, but did not provide a timeline for the fix. See case # 5814859491 if you have access to the histories.

The workaround for now has been change the resource name of the ASG in the template, forcing CF to create a new autoscaling group with the relevant configuration updates, and the old one is discarded.

**Comment by BugDestroy3r** [comment]
@mrapczynski have you got any update on this bug-fixes? 

trying to debug for hours until seeing this and the workaround is not realistic for the stack that frequently changed and/or under active development.

**Comment by mrapczynski** [comment]
@BugDestroy3r No, as far I know, this is still a problem. I do not work for AWS, and have been experiencing this as a customer.

**Comment by Trane9991** [comment]
So, I've reported this to AWS Support and they told me that they were able to reproduce the issue, but "could not provide an ETA on when the fix will be deployed" 🤷‍♂️
